src/app/components/sign-up/sign-up.component.html

```diff
@@ -1,139 +1,150 @@
-            id="username"
-            type="text"
-            formControlName="username"
-            placeholder="Username"
-            class="bg-transparent border-b-[1px] border-grey/60 h-8 w-full pl-2"
-            formControlName="username"
-          />
-        </div>
-        @if (signupForm.get('username')?.errors &&
-        signupForm.get('username')?.touched) {
-        <div class="text-red-500 text-center text-xs">
-          Username must be between 4 and 24 characters.
-        </div>
-        }
-      </div>
-      <div class="flex flex-col gap-1">
-        <div class="flex items-end w-full">
-          <div class="border-b-[1px] border-grey/60 pl-1">
-            <ng-icon name="heroUser" size="22" class="text-grey/60"></ng-icon>
+          @if (signupForm.get('username')?.errors &&
+          signupForm.get('username')?.touched) {
+          <div class="text-red-500 text-center text-xs">
+            Username must be between 4 and 24 characters.
           </div>
-          <input
-            id="email"
-            type="email"
-            formControlName="email"
-            placeholder="Email"
-            class="bg-transparent border-b-[1px] border-grey/60 h-8 w-full pl-2"
-          />
-        </div>
-        @if (signupForm.get('email')?.errors &&
-        signupForm.get('email')?.touched) {
-        <div class="text-red-500 text-center text-xs">
-          Email must be a valid email address.
+          }
         </div>
-        }
-      </div>
-      <div class="flex flex-col gap-1">
-        <div class="flex items-end w-full">
-          <div class="border-b-[1px] border-grey/60 pl-1">
-            <ng-icon
-              name="heroLockClosed"
-              size="22"
-              class="text-grey/60"
-            ></ng-icon>
+        <div class="flex flex-col gap-1">
+          <div class="flex items-end w-full">
+            <div class="border-b-[1px] border-white/60 pl-1">
+              <ng-icon
+                name="heroUser"
+                size="22"
+                class="text-white/60"
+              ></ng-icon>
+            </div>
+            <input
+              id="email"
+              type="email"
+              formControlName="email"
+              placeholder="Email"
+              class="bg-transparent border-b-[1px] border-white/60 h-8 w-full pl-2"
+            />
           </div>
-          <input
-            id="password"
-            type="{{ showPassword ? 'password' : 'text' }}"
-            formControlName="password"
-            placeholder="Password"
-            class="bg-transparent border-b-[1px] border-grey/60 h-8 w-full pl-2"
-          />
-          <div class="border-b-[1px] border-grey/60 pl-1">
-            <ng-icon
-              name="{{ showPassword ? 'heroEye' : 'heroEyeSlash' }}"
-              size="22"
-              class="text-grey/60"
-              (click)="togglePassword()"
-            ></ng-icon>
+          @if (signupForm.get('email')?.errors &&
```

Comment:
Good use of ` @if ` control flow to show the error message only once it is touched 👏